ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Basics.hs

```diff
@@ -484,30 +507,42 @@ forgetTickedLedgerStateTracking = mapOverLedgerTablesTicked valuesTrackingMK
   Concrete ledger tables
 -------------------------------------------------------------------------------}
 
+type MapKind         = {- key -} Type -> {- value -} Type -> Type
 type LedgerStateKind = MapKind -> Type
 
-data MapKind = DiffMK
-             | EmptyMK
-             | KeysMK
-             | QueryMK
-             | RewoundMK
-             | SeqDiffMK
-             | TrackingMK
-             | ValuesMK
-
-data ApplyMapKind :: MapKind -> Type -> Type -> Type where
-  ApplyDiffMK     :: !(UtxoDiff    k v)                    -> ApplyMapKind DiffMK       k v
-  ApplyEmptyMK    ::                                          ApplyMapKind EmptyMK      k v
-  ApplyKeysMK     :: !(UtxoKeys    k v)                    -> ApplyMapKind KeysMK       k v
-  ApplySeqDiffMK  :: !(SeqUtxoDiff k v)                    -> ApplyMapKind SeqDiffMK    k v
-  ApplyTrackingMK :: !(UtxoValues  k v) -> !(UtxoDiff k v) -> ApplyMapKind TrackingMK   k v
-  ApplyValuesMK   :: !(UtxoValues  k v)                    -> ApplyMapKind ValuesMK     k v
-  ApplyRewoundMK  :: !(RewoundKeys k v)                    -> ApplyMapKind RewoundMK    k v
-
-  ApplyQueryAllMK  ::                    ApplyMapKind QueryMK k v
-  ApplyQuerySomeMK :: !(UtxoKeys k v) -> ApplyMapKind QueryMK k v
-
-emptyAppliedMK :: Ord k => SMapKind mk -> ApplyMapKind mk k v
+data MapKind' = DiffMK'
+              | EmptyMK'
+              | KeysMK'
+              | QueryMK'
+              | RewoundMK'
+              | SeqDiffMK'
+              | TrackingMK'
+              | ValuesMK'
+
+type DiffMK     = ApplyMapKind' DiffMK'
+type EmptyMK    = ApplyMapKind' EmptyMK'
```

Comment:
I'm a bit worried about having the ticked and unticked versions of `ApplyMapKind` and the constructors of `MapKind'`. For instance `EmptyMK` actually represents an application of the map kind to a promoted constructor that is called`EmptyMK'`.  But I don't have an alternative so I'm just complaining about the Government without going to vote :)

Reply:
Maybe: 

```
type AppEmptyMK    = ApplyMapKind' EmptyMK
```

:ballot_box:

Reply:
Or we might just want to remove the extra level of indirection.